custom/aaa/tests/test_aggregations.py

```diff
@@ -0,0 +1,202 @@
+    'growth_monitoring': 'config_report_reach-test_reach-growth_monitoring_forms_66bc5792',
+    'household': 'config_report_reach-test_reach-household_cases_73b9e4e8',
+    'person': 'config_report_reach-test_reach-person_cases_26a9647f',
+    'village': 'config_report_reach-test_reach-village_location_569fb159',
+}
+INPUT_PATH = os.path.join(os.path.dirname(__file__), 'data', 'ucr_tables')
+OUTPUT_PATH = os.path.join(os.path.dirname(__file__), 'data', 'agg_tables')
+TEST_DOMAIN = 'reach-test'
+TEST_ENVIRONMENT = 'icds'
+
+
+@override_settings(SERVER_ENVIRONMENT='icds')
+class AggregationScriptTestBase(CSVTestCase):
+    always_include_columns = {'person_case_id'}
+
+    @classmethod
+    def setUpClass(cls):
+        super(AggregationScriptTestBase, cls).setUpClass()
+        _setup_ucr_tables()
+        update_child_table(TEST_DOMAIN)
+        update_child_history_table(TEST_DOMAIN)
+        update_woman_table(TEST_DOMAIN)
+        update_woman_history_table(TEST_DOMAIN)
+        update_ccs_record_table(TEST_DOMAIN)
+
+        for month in range(1, 3):
+            update_agg_awc_table(TEST_DOMAIN, date(2019, month, 1))
+            update_agg_village_table(TEST_DOMAIN, date(2019, month, 1))
+
+    @classmethod
+    def tearDownClass(cls):
+        _teardown_ucr_tables()
+        super(AggregationScriptTestBase, cls).tearDownClass()
+
+    def _convert_decimal_to_string(self, value):
+        """
+            Args:
+                value (decimal.Decimal)
+            Returns:
+                str
+            Converts scientific notation to decimal form if needed.
+            it's needed because in csv file all numbers are written in decimal form.
+            Here is an example why we can't simply apply str to decimal number
+                >>> str(Decimal('0.0000000'))
+                '0E-7'
+                >>> self._convert_decimal_to_string(Decimal('0.0000000'))
+                '0.0000000'
+        """
+        value_str = str(value)
+        p = re.compile('0E-(?P<zeros>[0-9]+)')
+        match = p.match(value_str)
+        if match:
+            return '0.{}'.format(int(match.group('zeros')) * '0')
+        else:
+            return value_str
+
+    def _load_data_from_db(self, table_cls, sort_key):
+        for row in table_cls.objects.order_by(*sort_key).values().all():
+            for key, value in list(row.items()):
+                if isinstance(value, date):
+                    row[key] = value.strftime('%Y-%m-%d')
+                elif isinstance(value, time):
+                    row[key] = value.strftime("%H:%M:%S.%f").rstrip('0').rstrip('.')
+                elif isinstance(value, six.integer_types):
+                    row[key] = str(value)
+                elif isinstance(value, (float, Decimal)):
+                    row[key] = self._convert_decimal_to_string(row[key])
+                elif isinstance(value, six.string_types):
```

Comment:
@emord Is there a reason for encoding to bytes here?  This appears to be failing in python 3:
https://api.travis-ci.org/v3/job/498933818/log.txt